tf_model/train_classifier.py

```diff
@@ -0,0 +1,265 @@
+            train_data,
+            train_labels,
+            test_size=0.15,
+            random_state=7,
+            stratify=train_labels)
+        
+        return train_data, train_labels, val_data, val_labels, test_data, test_labels
+    
+
+    def get_weights(self, dataset: np.ndarray):
+
+        class_weights = compute_class_weight('balanced', np.unique(dataset), dataset)
+        class_weights = dict(enumerate(class_weights))
+
+        return class_weights
+
+
+    def preprocess_tweet(self, tweet: str):
+    
+        remove_linebreak = tf.strings.regex_replace(tweet, '\n', '')
+        remove_tab = tf.strings.regex_replace(remove_linebreak, '\t', '')
+        replace_ampersand = tf.strings.regex_replace(remove_tab, '&amp;', 'and')
+        remove_url = tf.strings.regex_replace(replace_ampersand, 'http\S+', '')
+        remove_mention = tf.strings.regex_replace(remove_url, '@\w+ ', ' ')
+        remove_contraction = tf.strings.regex_replace(remove_mention, "'", '')
+        alpha_char = tf.strings.regex_replace(remove_contraction, '[^a-zA-Z.]', ' ')
+        remove_outer_whitespace = tf.strings.strip(alpha_char)
+        remove_extra_whitespace = tf.strings.regex_replace(remove_outer_whitespace, '\s{2,}', ' ')
+        
+        return tf.strings.lower(remove_extra_whitespace)
+    
+    
+    def create_vectorizer(self, dataset: np.ndarray) -> TextVectorization:
+        
+        vectorizer = TextVectorization(
+            max_tokens=self.vocab_size,
+            standardize=self.preprocess_tweet,
+            output_mode='int',
+            output_sequence_length=self.max_length,
+            name='vectorizer')
+        
+        vectorizer.adapt(dataset)
+
+        return vectorizer
+    
+    
+    def create_emb_matrix(self, vectorizer):
+        
+        vocab = vectorizer.get_vocabulary()
+        word_index = dict(zip(vocab, range(len(vocab))))
+        
+        emb_matrix = np.zeros((self.vocab_size, self.embed_dim))
+        for word, index in word_index.items():
+            vector = glove_vectors.get(word)
+            if vector is not None:
+                emb_matrix[index] = vector
+                
+        return emb_matrix
+        
+        
+    def call(self, inputs, training=False):
+        
+        data, labels = self.get_data()
+        train_data, train_labels, val_data, val_labels, _, _ = self.data_splits(data, labels)
+        vectorizer = self.create_vectorizer(data)
+        emb_matrix = self.create_emb_matrix(vectorizer)
+        
+        tweet_input = Input(shape=(1,), dtype=tf.string, name='tweet')
+        
+        x = vectorizer(tweet_input)
+        
+        x = Embedding(input_dim=self.vocab_size,
+                      output_dim=self.embed_dim,
+                      input_length=self.max_length,
+                      weights=[emb_matrix],
+                      trainable=False,
+                      name='GloVe_200')(x)
+        
+        x_1 = self.bigram(x)
```

Comment:
I find it a little hard to follow that the layers are defined in `__init__` and referenced here. is there any reason the logic can't be grouped together? also curious why you made each layer a class attribute